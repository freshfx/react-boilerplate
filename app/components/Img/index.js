/**
 *
 * Img.js
 *
 * Renders an image, enforcing the usage of the alt="" tag
 */

import React from 'react'
import PropTypes from 'prop-types'

const Img = props => <img className={props.className} src={props.src} alt={props.alt} />

// We require the use of src and alt, only enforced by react in dev mode
Img.propTypes = {
  alt: PropTypes.string.isRequired,
  className: PropTypes.string,
  src: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]).isRequired
}

Img.defaultProps = {
  className: null
}

export default Img
