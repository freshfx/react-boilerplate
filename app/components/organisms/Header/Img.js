import styled from '@emotion/styled'

import NormalImg from 'components/atoms/Img'

const Img = styled(NormalImg)`
  width: 100%;
  margin: 0 auto;
  display: block;
`

export default Img
