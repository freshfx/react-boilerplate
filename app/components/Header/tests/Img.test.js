import React from 'react'
import {render} from '@testing-library/react'

import Img from '../Img'

describe('Img', () => {
  it('should match snapshot', () => {
    const {container} = render(<Img src="http://example.com/test.jpg" alt="test" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
