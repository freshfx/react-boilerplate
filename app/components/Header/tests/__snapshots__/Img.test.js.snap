// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img should match snapshot 1`] = `
.emotion-0 {
  width: 100%;
  margin: 0 auto;
  display: block;
}

<img
  alt="test"
  class="emotion-0 emotion-1"
  src="http://example.com/test.jpg"
/>
`;
